sklearn/pipeline.py

```diff
@@ -160,17 +165,17 @@ def _validate_steps(self):
         for t in transformers:
             if t is None:
                 continue
-            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
-                    hasattr(t, "transform")):
+            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform"))
```

Comment:
There's lot's of unrelated formatting changes in this. it would be easier to review if these weren't here. Generally we try no to do those because they can lead to merge conflicts and provide little value.

Reply:
This PR has been updated to remove all these slight formatting changes.

Reply:
This PR has been updated by removing most formatting errors.